go/consensus/tendermint/staking/staking.go

```diff
@@ -242,35 +236,59 @@ func (tb *tendermintBackend) worker(ctx context.Context) {
 }
 
 func (tb *tendermintBackend) onEventDataNewBlock(ctx context.Context, ev tmtypes.EventDataNewBlock) {
-	events := tmapi.ConvertBlockEvents(ev.ResultBeginBlock.GetEvents(), ev.ResultEndBlock.GetEvents())
-
-	_, _ = tb.onABCIEvents(ctx, events, ev.Block.Header.Height, true)
+	tmEvents := append(ev.ResultBeginBlock.GetEvents(), ev.ResultEndBlock.GetEvents()...)
+	events, errs := EventsFromTendermintEvents(nil, ev.Block.Header.Height, tmEvents)
+	for _, err := range errs {
+		tb.logger.Error("error processing staking event", "err", err)
+	}
+	tb.notify(events)
 }
 
-func (tb *tendermintBackend) onEventDataTx(ctx context.Context, tx tmtypes.EventDataTx) {
-	evHash := hash.NewFromBytes(tx.Tx)
-
-	var events []tmapi.EventWithHash
-	for _, tmEv := range tx.Result.Events {
-		var ev tmapi.EventWithHash
-		ev.Event = tmEv
-		ev.TxHash = evHash
-		events = append(events, ev)
+func (tb *tendermintBackend) onEventDataTx(ctx context.Context, ev tmtypes.EventDataTx) {
+	events, errs := EventsFromTendermintEvents(ev.Tx, ev.Height, ev.Result.Events)
+	for _, err := range errs {
+		tb.logger.Error("error processing staking event", "err", err)
 	}
+	tb.notify(events)
+}
 
-	_, _ = tb.onABCIEvents(ctx, events, tx.Height, true)
+func (tb *tendermintBackend) notify(events []api.Event) {
+	for _, ev := range events {
+		if ev.Transfer != nil {
```

Comment:
I think we should consider removing these event-specific notifiers and Watch endpoints as they don't seem all that useful now that we have the general one. But that's probably for a different issue.

Reply:
Ok, i'll open an issue for it.